pkg/services/object/put/distributed.go

```diff
@@ -199,36 +204,86 @@ func (t *distributedTarget) sendObject(node nodeDesc) error {
 		return fmt.Errorf("could not write header: %w", err)
 	}
 
-	_, sig, err := target.Close()
+	_, sigsRaw, err := target.Close()
 	if err != nil {
 		return fmt.Errorf("could not close object stream: %w", err)
 	}
 
 	if t.localNodeInContainer && !node.local {
 		// These should technically be errors, but we don't have
 		// a complete implementation now, so errors are substituted with logs.
-		var l = t.placementIterator.log.With(zap.Stringer("oid", t.obj.GetID()))
+		var l = t.placementIterator.log.With(zap.Stringer("oid", t.obj.GetID()),
+			zap.String("node", network.StringifyGroup(node.info.AddressGroup())))
 
-		if sig == nil {
-			l.Info("missing object meta signature")
+		sigs, err := decodeSignatures(sigsRaw)
+		if err != nil {
+			l.Info("failed to decode signatures", zap.Error(err))
 			return nil
 		}
 
-		if !bytes.Equal(sig.PublicKeyBytes(), node.info.PublicKey()) {
-			l.Info("public key differs in object meta signature")
+		for i, sig := range sigs {
+			if !bytes.Equal(sig.PublicKeyBytes(), node.info.PublicKey()) {
+				l.Info("public key differs in object meta signature", zap.Int("signature index", i))
+				continue
+			}
+
+			if !sig.Verify(t.objSharedMeta) {
```

Comment:
@roman-khimov, hm, that is right. but i am not sure what else can be done here, we cannot collect two different signatures since we cannot send half signatures for the "outdated" nodes and half for the "actual" one, server-side always needs to get a single version for vub (or we need contract updates again). i can suggest sending three signatures and the server can always try `currentEpoch + 1` (btw, this is even better, since we can get an object in the last block of the epoch, and then handling may break in the next block on the contract side), which handles every kind of epoch divergence that is not bigger than 1 epoch